### PR TITLE
don't make script errors fatal error

### PR DESCRIPTION
**Pull request by mjkzy**

### CODE REVIEW

File: src/client/component/client_patches.cpp [hunk 1]
```diff
@@ -69,6 +69,20 @@ namespace client_patches
...
+		void script_errors_stub(const char* file, int line, unsigned int code, const char* fmt, ...)
+		{
+			char buffer[0x1000];
+
+			{
+				va_list ap;
+				va_start(ap, fmt);
+				vsnprintf_s(buffer, sizeof(buffer), _TRUNCATE, fmt, ap);
+				va_end(ap);
+			}
```
Comment (momo5502) [inline src/client/component/client_patches.cpp:82]: good practice before com error, not sure if intentional, but very good to scope

File: src/client/component/client_patches.cpp [hunk 2]
```diff
@@ -82,6 +96,9 @@ namespace client_patches
 			utils::hook::set(0x15AAEB254_g, mixer_open_stub);
 
 			preload_map_hook.create(0x14135A1E0_g, preload_map_stub);
+
+			// don't make script errors fatal error
+			utils::hook::call(0x1412CAC4D_g, script_errors_stub);
```
Comment (momo5502) [inline src/client/component/client_patches.cpp:101]: what about the dedi?

**Comment by mjkzy** [comment]
oh yeah, forgot to do dedi. will do that inna bit

File: src/client/component/client_patches.cpp (hunk 2 above)
Comment (mjkzy) [reply src/client/component/client_patches.cpp:101]: will add dedi support in a bit

File: src/client/component/patches.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+			char buffer[0x1000];
+
+			{
+				va_list ap;
+				va_start(ap, fmt);
+				vsnprintf_s(buffer, sizeof(buffer), _TRUNCATE, fmt, ap);
+				va_end(ap);
+			}
+
+			game::Com_Error(game::ERROR_SCRIPT_DROP, "%s", buffer);
```
Comment (momo5502) [inline src/client/component/patches.cpp:22]: the address for com_error for dedis is not yet added in symbols.hpp!

File: src/client/component/patches.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		}
+	}
+
+	struct component final : generic_component
+	{
+		void post_unpack() override
+		{
+			// don't make script errors fatal error
+			utils::hook::call(game::select(0x1412CAC4D_g, 0x140158EB2_g), script_errors_stub);
```
Comment (momo5502) [inline src/client/component/patches.cpp:31]: select doesn't require the _g literal

**Review by momo5502 — CHANGES_REQUESTED** [review]
i wish the formatting was in a separate PR, that would make it easier to review and properly separate the concerns

File: src/client/component/patches.cpp (hunk 3 above)
Comment (mjkzy) [reply src/client/component/patches.cpp:22]: ah poop i should have looked into this more, sorryyyy

**Review by momo5502 — APPROVED** [review]

**Pull request merged**